Client/Frontend/Browser/Tab.swift

```diff
@@ -435,27 +435,29 @@ class Tab: NSObject {
         _ = webView?.go(to: item)
     }
 
-    @discardableResult func loadRequest(_ request: URLRequest) -> WKNavigation? {
+    // Ecosia: adding async callback for navigation result to inject cookie
+    func loadRequest(_ request: URLRequest, completion: ((WKNavigation?) -> ())? = nil ) {
         if let webView = webView {
             // Convert about:reader?url=http://example.com URLs to local ReaderMode URLs
             if let url = request.url, let syncedReaderModeURL = url.decodeReaderModeURL, let localReaderModeURL = syncedReaderModeURL.encodeReaderModeURL(WebServer.sharedInstance.baseReaderModeURL()) {
                 let readerModeRequest = PrivilegedRequest(url: localReaderModeURL) as URLRequest
                 lastRequest = readerModeRequest
-                return webView.load(readerModeRequest)
+                completion?(webView.load(readerModeRequest))
             }
             lastRequest = request
             if let url = request.url, url.isFileURL, request.isPrivileged {
-                return webView.loadFileURL(url, allowingReadAccessTo: url)
+                completion?(webView.loadFileURL(url, allowingReadAccessTo: url))
             }
 
             // Ecosia: inject cookie
-            var request = request
             if let cookie = browserViewController?.tabManager.cookie {
+                var request = request
                 request.allHTTPHeaderFields = HTTPCookie.requestHeaderFields(with: [cookie.value])
```

Comment:
so as I see it injecting the cookie via `request.allHTTPHeaderFields ` will make it work for private mode but I think we need to use `webView.configuration.websiteDataStore.httpCookieStore.setCookie` for non private tabs to keep logged in state for new tabs. If not, we would not need the PR at all as it apparently does not change the way the cookie is injected or am I overseeing anything?